src/Illuminate/Validation/Validator.php

```diff
@@ -584,8 +584,12 @@ protected function shouldStopValidating($attribute)
      * @param  array   $parameters
      * @return void
      */
-    protected function addFailure($attribute, $rule, $parameters)
+    public function addFailure($attribute, $rule, $parameters = [])
     {
+        if (! $this->messages) {
+            $this->passes();
+        }
+
```

Comment:
I don't see any tests for this logic. What is this doing?

Reply:
We need a test for that logic to make sure it works either way.

Reply:
@tillkruss. Thank you.  I agree the tests are needed. My tests are asserting it (see section 2 of my answer). Please, specify what should I do?

Reply:
@tillkruss , I just checked one more time. My test cases are covering this functionality. If the code that you mention is deleted the test case 'testAddFailureIsFunctional'  is getting an error. `Error: Call to a member function add() on null`.

Reply:
I looked through other test cases, there are plenty of tests around `$validator->passes();`.